Sources/Kitura-CredentialsTwitter/CredentialsTwitter.swift

```diff
@@ -0,0 +1,350 @@
+                             inProgress: @escaping () -> Void) {
+        var requestOptions: [ClientRequest.Options] = []
+        requestOptions.append(.schema("https://"))
+        requestOptions.append(.hostname("api.twitter.com"))
+        requestOptions.append(.method("POST"))
+        requestOptions.append(.path("oauth/request_token"))
+        var headers = [String: String]()
+        var parameters = [String:String]()
+
+        let nonce = String.nonce
+        let timeStamp = Date().timeIntervalSince1970.roundedString
+
+        parameters["oauth_consumer_key"] = consumerKey
+        parameters["oauth_signature_method"] = "HMAC-SHA1"
+        parameters["oauth_timestamp"] = timeStamp
+        parameters["oauth_nonce"] = nonce
+        parameters["oauth_version"] = "1.0"
+
+        guard let signature = String.oAuthSignature(fromMethod: "POST",
+                                                    urlString: "https://api.twitter.com/oauth/request_token",
+                                                    parameters: parameters,
+                                                    consumerSecret: consumerSecret) else {
+                                                        onFailure(nil, nil)
+                                                        return
+        }
+
+        parameters["oauth_signature"] = signature
+
+        headers["Authorization"] = String.oAuthAuthorizationString(fromParameters: parameters)
+        requestOptions.append(.headers(headers))
+
+        let twitterRequest = HTTP.request(requestOptions) {
+            (optionalResponse) in
+            guard let tokenResponse = optionalResponse, tokenResponse.statusCode == HTTPStatusCode.OK else {
+                onFailure(optionalResponse?.statusCode, nil)
+                return
+            }
+
+            var body = Data()
+            do {
+                try tokenResponse.readAllData(into: &body)
+                let string = String(data: body, encoding: String.Encoding.utf8)
+                guard let fields = string?.components(separatedBy: "&") else {
+                    onPass(tokenResponse.statusCode, .none)
+                    return
+                }
+                var responseDictionary = [String: String]()
+                for field in fields {
+                    let keyValue = field.components(separatedBy: "=")
+                    if let key = keyValue.first,
+                        let value = keyValue.last {
+                        responseDictionary[key] = value
+                    }
+                    else {
+                        onPass(tokenResponse.statusCode, .none)
+                        return
+                    }
+                }
+
+                guard let token = responseDictionary["oauth_token"],
+                    let secret = responseDictionary["oauth_token_secret"] else {
+                        onFailure(nil, nil)
```

Comment:
Comparing with other completion blocks above, what makes this an `onFailure` instead of `onPass`?